src/Products/PluginIndexes/common/UnIndex.py

```diff
@@ -324,6 +329,65 @@ def _apply_not(self, not_parm, resultset=None):
     def _convert(self, value, default=None):
         return value
 
+    def getRequestCache(self):
+        """returns dict for caching per request for interim results
+        of an index search. Returns 'None' if no REQUEST attribute
+        is available"""
+        cache = None
+        REQUEST = aq_get(self, 'REQUEST', None)
+        if REQUEST is not None:
+            catalog = aq_parent(aq_parent(aq_inner(self)))
+            if catalog is not None:
+                key = self._catalog_cache_key(catalog)
+                cache = REQUEST.get(key, None)
+                if cache is None:
+                    cache = REQUEST[key] = RequestCache()
+
+        return cache
+
+    def _catalog_cache_key(self, catalog):
+        # unique catalog identifier
+        cid = '_%s_%s' % (catalog.getId(), id(catalog))
```

Comment:
I'd add a constant prefix to this, like `_catalogcache_%s_%s` % (...)` - just to make it easier to spot what this is when looking at a request object. It would also lower the chance of key conflicts on the request object.